cartoframes/data/utils.py

```diff
@@ -290,9 +292,86 @@ def is_table_name(data):
     return isinstance(data, str) and normalize_name(data) == data
 
 
-def _save_index_as_column(df):
+def save_index_as_column(df):
     index_name = df.index.name
     if index_name is not None:
         if index_name not in df.columns:
             df.reset_index(inplace=True)
             df.set_index(index_name, drop=False, inplace=True)
+
+
+def get_query_geom_type(context, query):
+    """Fetch geom type of a remote table"""
+    response = context.execute_query('''
+        SELECT distinct ST_GeometryType(the_geom) AS geom_type
+        FROM ({}) q
+        LIMIT 5
+    '''.format(query), do_post=False)
+    if response and response.get('rows') and len(response.get('rows')) > 0:
+        st_geom_type = response.get('rows')[0].get('geom_type')
+        if st_geom_type:
+            return map_geom_type(st_geom_type[3:])
+
+
+# Country names are pegged to the following query:
+# SELECT
+#    count(*) num_measurements,
+#    tag.key region_id,
+#    tag.value region_name
+#  FROM (
+#    SELECT *
+#    FROM OBS_GetAvailableNumerators()
+#    WHERE jsonb_pretty(numer_tags) LIKE '%subsection/%'
+#  ) numers,
+#  Jsonb_Each(numers.numer_tags) tag
+#  WHERE tag.key like 'section%'
+#  GROUP BY tag.key, tag.value
+#  ORDER BY region_name
+
+
+REGIONTAGS = {
+    'Australia': 'section/tags.au',
+    'Brazil': 'section/tags.br',
+    'Canada': 'section/tags.ca',
+    'European Union': 'section/tags.eu',
+    'France': 'section/tags.fr',
+    'Mexico': 'section/tags.mx',
+    'Spain': 'section/tags.spain',
+    'United Kingdom': 'section/tags.uk',
+    'United States': 'section/tags.united_states'
+}
+
+
+def get_countrytag(country):
```

Comment:
I think this (and `REGIONTAGS`) should be in the same namespace than Data Observatory class, in a utils file or inside the module. Dataset and related stuff is not going to use it because it is something adhoc of DO